hadoop-ozone/s3gateway/src/test/java/org/apache/hadoop/ozone/s3/endpoint/EndpointBuilder.java

```diff
@@ -74,6 +76,11 @@ public EndpointBuilder<T> setRequestId(RequestIdentifier newRequestId) {
     return this;
   }
 
+  public EndpointBuilder<T> setSignatureInfo(SignatureInfo newSignatureInfo) {
+    this.signatureInfo = newSignatureInfo;
+    return this;
+  }
```

Comment:
How about we set

```java
SignatureInfo signatureInfo = mock(SignatureInfo.class);
when(signatureInfo.isSignPayload()).thenReturn(true);
```
as a default value `SignatureInfo` to the `EndpointBuilder` (similar to `OzoneConfiguration` and `RequestIdenfitifer`). This way, we don't need to add codes in each of the tests (in the spirit of HDDS-12116).